packages/host/app/resources/metamask.ts

```diff
@@ -0,0 +1,111 @@
+import { tracked } from '@glimmer/tracking';
+import { Resource } from 'ember-resources/core';
+import { enqueueTask, restartableTask } from 'ember-concurrency';
+
+// window is a global object that is available in this resource
+declare global {
+  interface Window {
+    ethereum: any;
+  }
+}
+
+const METAMASK_ERROR_CODES = {
+  user_rejected: 4001,
+  unknown_chain: 4902,
+};
+
+class MetaMaskResource extends Resource {
```

Comment:
We should pair on this. There are not clear docs on how to build a good resource so it's not obvious to discover by oneself.